pkg/virt-launcher/virtwrap/agent-poller/agent_poller.go

```diff
@@ -342,3 +309,62 @@ func (p *AgentPoller) Stop() {
 		p.agentDone = nil
 	}
 }
+
+// With libvirt 5.6.0 direct call to agent can be replaced with call to libvirt Domain.GetGuestInfo
+func executeAgentCommands(commands []AgentCommand, con cli.Connection, agentStore *AsyncAgentStore, domainName string) {
```

Comment:
can you please move this refactor (the part that just put it in a function) it to a separate commit ? 
as it moves the function, it will help focusing the 4th commit main logic